### ISSUE TITLE
Separate sugared and core statements in the Substance AST 

### ISSUE THREAD

**Issue by wodeni** [penrose/penrose#282]
@hypotext and I discussed the issue of where to place statements like `DeclList` and `DeclBind` in the Substance AST (#267). Our conclusion is that since they are both sugared constructs that decompose into multiple "core" Substance statements, it makes more sense to group them in a different type (named `Substmt'` for now) to make sure that they do not interfere with the future analysis on the core AST. 

We plan to implement this change after #267 and #278 are merged into `master`.

**Comment by wodeni** [comment]
Done. Closing the issue
